internal/config/validator_test.go

```diff
@@ -28,3 +29,255 @@ func TestValidator(t *testing.T) {
+								{
+									Communities: []string{
+										"myCommunityAlias",
+										"abcd",
+										"11111:11aaaa",
+										"larg:12345:12345:12345",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectclusterResources: ClusterResources{
+				Peers: []metallbv1beta2.BGPPeer{
+					{
+						Spec: metallbv1beta2.BGPPeerSpec{
+							BFDProfile: "",
+						},
+					},
+					{
+						Spec: metallbv1beta2.BGPPeerSpec{
+							PasswordSecret: v1.SecretReference{
+								Name: "",
+							},
+						},
+					},
+				},
+				Communities: []metallbv1beta1.Community{
+					{
+						Spec: metallbv1beta1.CommunitySpec{
+							Communities: []metallbv1beta1.CommunityAlias{
+								{
+									Name:  "myCommunityAlias",
+									Value: "65000:100",
+								},
+							},
+						},
+					},
+				},
+				BGPAdvs: []metallbv1beta1.BGPAdvertisement{
+					{
+						Spec: metallbv1beta1.BGPAdvertisementSpec{
+							Communities: []string{
+								"myCommunityAlias",
+								"11111:11aaaa",
+								"larg:12345:12345:12345",
+							},
+						},
+					},
+				},
+				LegacyAddressPools: []metallbv1beta1.AddressPool{
+					{
+						Spec: metallbv1beta1.AddressPoolSpec{
+							BGPAdvertisements: []metallbv1beta1.LegacyBgpAdvertisement{
+								{
+									Communities: []string{
+										"myCommunityAlias",
+										"11111:11aaaa",
+										"larg:12345:12345:12345",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		result := resetTransientErrorsFields(test.clusterResources)
+		for i := range result.Peers {
+			if result.Peers[i].Spec.BFDProfile != "" && result.Peers[i].Spec.PasswordSecret.Name != "" {
+				t.Errorf("Expect that BFDProfile is %v and Name of the PasswordSecret is %v , but result BFDProfile is %v and Name of the PasswordSecret is %v",
+					test.expectclusterResources.Peers[i].Spec.BFDProfile, test.expectclusterResources.Peers[i].Spec.PasswordSecret.Name,
+					result.Peers[i].Spec.BFDProfile, result.Peers[i].Spec.PasswordSecret.Name)
+			}
+		}
+		for i := range result.BGPAdvs {
+			if !compareStringArrays(result.BGPAdvs[i].Spec.Communities, test.expectclusterResources.BGPAdvs[i].Spec.Communities) {
+				t.Errorf("Expect is %v, but result is %v", test.expectclusterResources.BGPAdvs[i].Spec.Communities,
+					result.BGPAdvs[i].Spec.Communities)
+			}
+		}
+		for i := range result.LegacyAddressPools {
+			for j := range result.LegacyAddressPools[i].Spec.BGPAdvertisements {
+				if !compareStringArrays(result.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities,
+					test.expectclusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities) {
+					t.Errorf("Expect is %v, but result is %v", test.expectclusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities,
+						result.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities)
+				}
+			}
+		}
```

Comment:
this can be replaced by:
```go
			if !cmp.Equal(result, expected) {
				t.Fatalf("test %s failed, unexpected clusterResources (-want +got)\n", test.desc, cmp.Diff(result, expected))
			}
```

Reply:
Thank you!! I searched for a function that does it, but I didn't find one. So, I wrote all of this. I will change it. tnx!